js/script.js

```diff
@@ -0,0 +1,155 @@
+    cards_container.className = "container";
+    // building cards
+    for(const item of courses){
+        if(search != undefined) course = item;
+        if(item == course){
+            for(const item in data.record[course].courses){
+                if(search != undefined){
+                    let title = data.record[course].courses[item].title;
+                    title = title.toLowerCase()
+                    search = search.toLowerCase()
+                    if(!title.includes(search)) continue;
+                }
+                
+                let instructor = "";
+                for(const instr in data.record[course].courses[item].instructors){
+                    instructor += data.record[course].courses[item].instructors[instr].name + ', ';
+                }
+                instructor = instructor.slice(0, -2);
+
+                cardlink = document.createElement("a");
+                cardlink.href = "";
+                cards_container.className = "container";
+                datacontainer = document.createElement("div");
+                datacontainer.className = "course";
+
+                image = document.createElement("img");
+                image.src = data.record[course].courses[item].image;
+                image.alt = "course thumbnail";
+                image.className = "thumb";
+                
+                heading = document.createElement("h4");
+                headingtxt = document.createTextNode(data.record[course].courses[item].title);
+                heading.appendChild(headingtxt);
+                
+                names = document.createElement("p")
+                namestxt = document.createTextNode(instructor);
+                names.appendChild(namestxt);
+
+                ratingcontainer = document.createElement("div");
+                ratingcontainer.className = "rating";
+                
+                rate = document.createElement("span");     
+                ratetxt = document.createTextNode(data.record[course].courses[item].rating.toFixed(2) + " "); 
+                rate.appendChild(ratetxt);
+
+                star1 = document.createElement("span");
+                star1.className = "fa fa-star";
+                star2 = document.createElement("span");
+                star2.className = "fa fa-star";
+                star3 = document.createElement("span");
+                star3.className = "fa fa-star";
+                star4 = document.createElement("span");
+                star4.className = "fa fa-star";
+                star5 = document.createElement("span");
+                star5.className = "fa fa-star-half-stroke";
+                ratingcontainer.appendChild(rate);
+                ratingcontainer.appendChild(star1);
+                ratingcontainer.appendChild(star2);
+                ratingcontainer.appendChild(star3);
+                ratingcontainer.appendChild(star4);
+                ratingcontainer.appendChild(star5);
```

Comment:
the structure of the stars is not dynamic. you should render the number of the stars depending on the rating number.

Reply:
i fixed it now but i altered the html, css and js files so i will not be able to fix it online on github and i don't have the raw code of phase 2 locally because i applied phase 3 requirements on it so you can find that this is fixed in phase 3 code.